src/databricks/labs/mcp/base.py

```diff
@@ -0,0 +1,147 @@
+"""
+Collection of base utils for MCP servers.
+"""
+
+import contextlib
+import logging
+from collections import deque
+from dataclasses import dataclass
+from typing import AsyncIterator
+from uuid import uuid4
+from starlette.applications import Starlette
+from starlette.routing import Mount
+from starlette.types import Receive, Scope, Send
+from mcp.server.streamable_http_manager import StreamableHTTPSessionManager
+
+from mcp.server import Server
+
+from mcp.server.streamable_http import (
+    EventCallback,
+    EventId,
+    EventMessage,
+    EventStore,
+    StreamId,
+)
+from mcp.types import JSONRPCMessage
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class EventEntry:
+    """
+    Represents an event entry in the event store.
+    """
+
+    event_id: EventId
+    stream_id: StreamId
+    message: JSONRPCMessage
+
+
+class InMemoryEventStore(EventStore):
```

Comment:
Curious if we actually need any in-memory event storage for our server, it is stateless right?

Reply:
I think we can keep memory for now, and remove it in future if needed. Apps have 6GB ram, ponentially we can also save into temp db